test/offsetarray.jl

```diff
@@ -313,7 +313,7 @@ seek(io, 0)
 @test readdlm(io, eltype(A)) == parent(A)
 
 amin, amax = extrema(parent(A))
-@test clamp(A, (amax+amin)/2, amax) == clamp(parent(A), (amax+amin)/2, amax)
+@test clamp.(A, (amax+amin)/2, amax).parent == clamp.(parent(A), (amax+amin)/2, amax)
```

Comment:
The existing vectorized `clamp` methods return an `Array` subtype independent of the input type (for example, `clamp`ing a `SparseMatrixCSC` or `OffsetArray` yields an `Array`). On the other hand `broadcast` better preserves the input type (for example, `clamp`ing a `SparseMatrixCSC` yields a `SparseMatrixCSC` and the analog holds for `OffsetArray`s). Thoughts? Thanks!